v2/event/event_unmarshal.go

```diff
@@ -359,7 +359,8 @@ func consumeDataAsBytes(e *Event, isBase64 bool, b []byte) error {
 	}
 
 	mt, _ := e.Context.GetDataMediaType()
-	if mt != ApplicationJSON && mt != TextJSON {
+	// Empty content type assumes json
+	if mt != "" && mt != ApplicationJSON && mt != TextJSON {
```

Comment:
This will throw new errors when the body is just an unquoted string or random bytes.

Reply:
Why? Can you provide a test case where do you think this is going to fail?